src/api/queueMarkAsRead.js

```diff
@@ -30,12 +30,13 @@ const processQueue = (auth: Auth) => {
     return;
   }
 
-  messagesFlags(auth, unsentMessageIds, 'add', 'read');
-  unsentMessageIds = [];
+  messagesFlags(auth, unackedMessageIds, 'add', 'read').then(success => {
+    unackedMessageIds = unackedMessageIds.filter(id => !success.messages.includes(id));
+  });
```

Comment:
We've more often used `async`/`await` syntax for Promises than `.then`, etc., but I'm not sure if that's something we'd like to enforce or if it just ends up being convenient most of the time. One nice benefit is its use in combination with `try`/`catch`, and we don't have `catch`-like logic here.

If we wanted to do that, one way is to make `processQueue` itself `async`, which
- allows the use of `await` in its body, and
- tells callers that it returns a Promise (hopefully a Promise that some callers might be interested in `await`ing). 

Then we could do,

```suggestion
  const responsePromise = messagesFlags(auth, unackedMessageIds, 'add', 'read');
  lastSentTime = Date.now();
  const success = await responsePromise;
  unackedMessageIds = unackedMessageIds.filter(id => !success.messages.includes(id));
```

or perhaps more directly,

```suggestion
  lastSentTime = Date.now();
  const success = await messagesFlags(auth, unackedMessageIds, 'add', 'read');
  unackedMessageIds = unackedMessageIds.filter(id => !success.messages.includes(id));
```

But I think I'm not fussy about using `async`/`await` here; as I'll mention in my top-level review comment, this isn't the code we plan to settle on, and making `processQueue` `async` is a change to its interface that we may or may not like.

(Another way to use `async`/`await`, one that doesn't mean making `processQueue` `async`, would be to use an `async` IIFE, but I don't do this much and I don't think it's clearer than `.then`. E.g.,

```js
  lastSentTime = Date.now();
  (async () => {
    const success = await messagesFlags(auth, unackedMessageIds, 'add', 'read');
    unackedMessageIds = unackedMessageIds.filter(id => !success.messages.includes(id));
  })();
```

)

Reply:
That makes sense. My preference is to leave this as `.then` for now, and figure out whether we actually want an async API or not in the future, as I adjust and refactor this code more. (although I'm mostly familiar with Rust `async`/`await`, not javascript, so my intuition could be off here).

Let me know if you'd rather I go in a specific direction in this PR.

Reply:
The async logic here is pretty simple and I don't have a strong preference on whether it's written with `async`/`await` or with `.then`.

In general, though, I think we'll generally prefer `await` when things get at all complicated -- in that case it keeps things a lot easier to follow and understand than the corresponding `.then`/`.catch`/etc code.

Detailed discussion in chat:
https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/async/near/1160961

Reply:
In further discussion there, Anders convinced me that we should use `await` rather than `.then`, here and systematically. I've filed #4652 for the handful of existing spots in general, but in particular it'd be a good followup to this PR to do so here.